File: api/src/main/java/tutmirleid/api/entity/ScenarioStepEntity.java
package tutmirleid.api.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;

@Entity
@Table(name = "scenario_step")
public class ScenarioStepEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String step;

    @Column(nullable = false)
    @JsonProperty("expectedResult")
    private String expectedResult;

    @ManyToOne
    @JoinColumn(name = "scenario_id")
    private ScenarioEntity scenario;

    public ScenarioStepEntity() {}

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getStep() {
        return step;
    }

    public void setStep(String step) {
        this.step = step;
    }

    public String getExpectedResult() {
        return expectedResult;
    }

    public void setExpectedResult(String expectedResult) {
        this.expectedResult = expectedResult;
    }

    public ScenarioEntity getScenario() {
        return scenario;
    }

    public void setScenario(ScenarioEntity scenario) {
        this.scenario = scenario;
    }
}

File: src/entity/scenario.step.entity.ts
import {
    Column,
    Entity,
    ManyToOne,
    PrimaryGeneratedColumn,
    JoinColumn,
} from 'typeorm';
import { Expose } from 'class-transformer';
import {ScenarioEntity} from "./scenario.entity";

@Entity({ name: 'scenario_step' })
export class ScenarioStepEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({
        type: 'varchar',
        length: 255,
        name: 'step',
    })
    step?: string;

    @Column({
        type: 'varchar',
        length: 255,
        name: 'expected_result',
    })
    @Expose({ name: 'expected_result' })
    expectedResult?: string;

    @ManyToOne(() => ScenarioEntity, scenario => scenario.steps, {
        onDelete: 'CASCADE',
    })
    @JoinColumn({ name: 'scenario_id' })
    @Expose({ name: 'scenario_id' })
    scenario: ScenarioEntity;
}
